h2/stream.py

```diff
@@ -1050,6 +1055,11 @@ def _initialize_content_length(self, headers):
         _expected_content_length field from it. It's not an error for no
         Content-Length header to be present.
         """
+
+        if self.request_method in (b'HEAD', u'HEAD'):
```

Comment:
If we make the change suggested in `extract_method_header`, this can become a straight bytestring comparison.